tla/consensus/ccfraft.tla

```diff
@@ -43,21 +43,39 @@ CONSTANTS
     \* from another node.
     None
 
-\* Membership states
-CONSTANTS
-    Active,
-    Retired
-
 LeadershipStates == {
     Follower,
     Candidate,
     Leader,
     None
     }
 
+\* Membership states, defined in ../src/kv/kv_types.h
+\* In this specification, the membership state of Retired also describes the
+\* retirement phase (also defined in ../src/kv/kv_types.h) to avoid the need for an additional variable
+CONSTANTS
+    \* Node is not in the process of leaving the CCF service.
+    Active,
+    \* Node has added its own retirement to its log, but it is not yet committed or even signed
+    \* The node can still revert to Active upon rollback
+    \* RetirementOrdered does not change the behavior of the node
+    RetirementOrdered,
+    \* Node has added its own retirement to its log and it has been signed but it is not yet committed
+    \* The node can still revert to RetirementOrdered or Active upon rollback
+    \* If the node is a leader, it will stop accepting new client requests
+    RetirementSigned,
+    \* Node retirement has been committed and it is no longer part of the network
+    \* If this node was a leader, it will step down. It will not run for election again.
```

Comment:
There isn't really an explicit "step down", other than sending a ProposeRequestVote to the most plausible successor. But otherwise it just stops sending heartbeats.

Reply:
Isn't this line the node stepping down?https://github.com/microsoft/CCF/blob/aeddbfb99d42a60fe26658fa12e1001c8d88c6f7/src/consensus/aft/raft.h#L1964

Reply:
Yes, in terms of internal state, you are right. I was thinking in terms of observable behaviour/messages for the other nodes.